File: byteback-specification/src/main/java/byteback/specification/ghost/HashMapSpec.java
package byteback.specification.ghost;

import byteback.specification.Contract.Abstract;
import byteback.specification.Contract.Behavior;
import byteback.specification.Contract.Invariant;
import byteback.specification.Contract.Return;
import byteback.specification.ghost.Ghost.Attach;

@Attach("java.util.HashMap")
@Invariant("is_mutable")
public class HashMapSpec<K, V> {

	@Behavior
	public boolean is_mutable() {
		return Ghost.of(MapSpec.class, this).is_mutable();
	}

	@Abstract
	@Return
	public HashMapSpec() {
	}

}

File: byteback-test/test-scala-2.13.8/src/main/scala/byteback/test/conversion/HashMapSpec.scala
package byteback.test.conversion;

import byteback.specification.Contract._
import byteback.specification.ghost._;
import byteback.specification.ghost.Ghost._

import collection.JavaConverters._
import collection.mutable._

@Attach("scala.collection.mutable.HashMap$")
abstract class HashMapSpec[K, V] {

  @Return
  def empty(): HashMap[K, V]

}
